### ISSUE TITLE
ignore_body_changes does not work with non properties in the body

### ISSUE THREAD

**Issue by emoshaya** [Azure/terraform-provider-azapi#349]
I am currently utilising the azapi_resource resource for the type ["Microsoft.Compute/virtualMachineScaleSets@2023-07-01"](https://learn.microsoft.com/en-us/azure/templates/microsoft.compute/virtualmachinescalesets?pivots=deployment-language-terraform)

As you can see in the documentation, the sku block is inside the body and I need to ignore the capacity value in the sku in the body. Here is the code:

```
resource "azapi_resource" "virtualMachineScaleSet" {
  type      = "Microsoft.Compute/virtualMachineScaleSets@2023-03-01"
  parent_id = azapi_resource.resourceGroup.id
  name      = var.resource_name
  location  = var.location
  body = jsonencode({
    identity = {
      type = "None"
    }
    properties = {
      additionalCapabilities = {
      }
      doNotRunExtensionsOnOverprovisionedVMs = false
      orchestrationMode                      = "Uniform"
      overprovision                          = true
      virtualMachineProfile = {
        diagnosticsProfile = {
          bootDiagnostics = {
            enabled    = false
            storageUri = ""
          }
        }  
        priority = "Regular"
      }
    }
    sku = {
      capacity = 0
      name     = "Standard_F2"
      tier     = "Standard"
    }
  })
  schema_validation_enabled = true
  response_export_values = null
  locks = null
  ignore_casing = false.
  ignore_body_changes = ["sku.capacity"]
  ignore_missing_property = true
}
```

despite me adding "sku.capacity" for ignore_body_changes, the plan still produces a plan summary for sky.capacity in the body

```
resource "azapi_resource" "virtualMachineScaleSet" {
      ~ body                      = jsonencode(
          ~ {
              ~ sku        = {
                  ~ capacity = 1 -> 0
                    name     = "Standard_F2"
                    # (1 unchanged attribute hidden)
                }
                # (1 unchanged attribute hidden)
            }
        )
        id                        = "/subscriptions/5e3845d8-000-0000-000-4e3758521888/resourceGroups/rg/providers/Microsoft.Compute/virtualMachineScaleSets/virtualMachineScaleSet"
      ~ ignore_body_changes       = [
          + "sku.capacity",
        ]
        name                      = "virtualMachineScaleSet"
      ~ output                    = jsonencode({}) -> (known after apply)
        # (7 unchanged attributes hidden)

        # (1 unchanged block hidden)
    }
```

**Comment by ms-henglu** [comment]
Hi @emoshaya ,

Thank you for taking time to report this issue!

The `ignore_body_changes` must be applied(don't worry, during this apply, the ignored fields won't be changed in the remote), and next `terraform plan` command will show there're no changes.

**Comment by e-moshaya** [comment]
> Hi @emoshaya ,
> 
> 
> 
> Thank you for taking time to report this issue!
> 
> 
> 
> The `ignore_body_changes` must be applied(don't worry, during this apply, the ignored fields won't be changed in the remote), and next `terraform plan` command will show there're no changes. 

Hi

Unfortunately that didn't work. The next plan also wanted to apply the change to the sku.capacity

**Comment by ms-henglu** [comment]
Hi @e-moshaya ,

We recently released v1.13.1 which supports dynamic properties, and it enables the native lifecycle.ignore_changes feature. Would you please give a try? Thanks!

More details: https://techcommunity.microsoft.com/t5/azure-tools-blog/announcing-azapi-dynamic-properties/ba-p/4121855

**Comment by ms-henglu** [comment]
Close this issue as it's completed, but feel free to reopen it if there's any question.
